php-checks/src/main/java/org/sonar/php/checks/utils/ReadWriteUsages.java

```diff
@@ -83,7 +85,12 @@ public void visitVariableDeclaration(VariableDeclarationTree tree) {
 
     @Override
     public void visitAssignmentExpression(AssignmentExpressionTree tree) {
-      visitAssignedVariable(tree.variable());
+      if (tree.getParent() instanceof ExpressionTree && !tree.operator().startsWith("=")) {
```

Comment:
I think the test on the parent should be improved. The code example on the [ticket](https://jira.sonarsource.com/browse/SONARPHP-831) itself still exhibits a FP because the parent of the compound assignment is a return statement.